app/views/contexts/calculated_summary_context.py

```diff
@@ -298,20 +298,34 @@ def _get_answer_format(self, groups: Iterable[Mapping]) -> Tuple[dict, list]:
                             "unit": answer.get("unit"),
                             "unit_length": answer.get("unit_length"),
                             "currency": answer.get("currency"),
+                            "decimal_places": 0,
                         }
+
+                        self._set_decimal_places(answer, answer_format)
+
                     answer_value = answer.get("value") or 0
                     values_to_calculate.append(answer_value)
 
         return answer_format, values_to_calculate
 
+    @staticmethod
+    def _set_decimal_places(answer: Mapping, answer_format: dict) -> None:
+        if decimal_places := answer.get("decimal_places"):
+            if decimal_places > answer_format["decimal_places"]:
+                answer_format["decimal_places"] = decimal_places
+
     @staticmethod
     def _format_total(
         *,
-        answer_format: Mapping[str, Literal["short", "long", "narrow"]],
+        answer_format: Mapping,
         total: NumericType,
     ) -> str:
         if answer_format["type"] == "currency":
-            return get_formatted_currency(total, answer_format["currency"])
+            return get_formatted_currency(
+                value=total,
+                currency=answer_format["currency"],
+                decimal_limit=answer_format.get("decimal_places") or 0,
```

Comment:
The `or 0` is redundant?. ``custom_format_currency` already handles the None.

Reply:
Updated 👍